src/citrine/resources/dataset.py

```diff
@@ -250,7 +250,7 @@ def register_all(self,
 
     def update(self, model: DataConcepts) -> DataConcepts:
         """Update a data model object using the appropriate collection."""
-        return self.gemd.update(model)
+        return self.gemd._collection_for(model).update(model)
```

Comment:
All DataConcepts update calls should route through the particular collection.

Reply:
why?